packages/next/server/web/adapter.ts

```diff
@@ -27,14 +27,34 @@ export async function adapter(params: {
   })
 
   const event = new NextFetchEvent({ request, page: params.page })
-  const original = await params.handler(request, event)
+  const response = await params.handler(request, event)
 
   return {
-    response: original || NextResponse.next(),
+    response: response || NextResponse.next(),
     waitUntil: Promise.all(event[waitUntilSymbol]),
   }
 }
 
+export function blockUnallowedResponse(
+  promise: Promise<FetchEventResult>
+): Promise<FetchEventResult> {
+  return promise.then((result) => ({
+    ...result,
+    response: result.response?.body
+      ? new Response(
+          JSON.stringify({
+            message: `A middleware can not alter response's body. Learn more: https://nextjs.org/docs/messages/returning-response-body-in-middleware`,
```

Comment:
I think we might want to `console.error()` this message so the end user doesn't see this response and then have the response only show `Internal Server Error` for the end user.

Reply:
Looks good other than this so after that's updated we can land